sources/PTYTextView.m

```diff
@@ -1434,7 +1435,8 @@ - (void)keyDown:(NSEvent*)event {
         // track the instance of PTYTextView that is currently handling a key event and rerouting
         // calls as needed in -insertText and -doCommandBySelector.
         gCurrentKeyEventTextView = [[self retain] autorelease];
-        [self interpretKeyEvents:[NSArray arrayWithObject:event]];
+        _keydownEvent = event;
+        _inputMethodIsInserting = !event.isARepeat && [self.inputContext handleEvent:event];
```

Comment:
Sure.

Follows are the steps exactly I did.

* Repository: git@github.com:tai2/iTerm2.git
* OSX 10.11.4
* Input source is preinstalled `U.S.`
* Build target is `iTerm2`

1. git checkout https://github.com/tai2/iTerm2/commit/cfdceb9b1da7d4e0dafe1f7db937b7fd94167f3c (use interpretKeyEvents)
2. `Clean` and `Run`.
3. Hold `k` in bash, it puts characters repeatedly.
4. git checkout https://github.com/tai2/iTerm2/commit/b8cf2bc0284fab5bef06d481e90ec0e68a1f6313 (use handleEvent)
5. `Clean` and `Run`.
6. Hold `k` in bash, **just one character is put.**
7. git checkout https://github.com/tai2/iTerm2/commit/5dad811de6c1d1dae466299531e1edce402d3cb8 (use handleEvent if not event.isARepeat)
8. `Clean` and `Run`.
9. Hold `k` in bash, the the character repeats.

The key `k` is just an example, but here is a strange thing.
In step 5, some keys didn't repeat but others did.

Unrepeatable characters: bdhjkmpqrtvwx01234567890
Repeatable characters: acefgilnosuyz`~!@#$%^&*()-_=+[]{}\|;:'",<.>/?

I hope this information gives you any hint.
Anyway, insert a check for event.isARepeat, then things do right as I expect.

interpretKeyEvents implementation you showed has a return pass, so handleEvent might not been called in our case.